### PR TITLE
Move EventSource to `OC` namespace

### PR DESCRIPTION
**Pull request by come-nc**
## Summary

There is an interface for it and a factory in OCP, so there’s no need to keep it in legacy folder.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by kesselb — APPROVED** [review]

File: lib/private/EventSourceFactory.php [hunk 1]
```diff
@@ -27,20 +30,12 @@
...
+	public function __construct(
+		private IRequest $request,
+	) {
 	}
 
-	/**
-	 * Create a new event source
-	 *
-	 * @return IEventSource
-	 * @since 28.0.0
```
Comment (susnux) [inline lib/private/EventSourceFactory.php:41]: why did you remove the comment?

**Review by susnux — APPROVED** [review]

File: lib/private/EventSourceFactory.php (hunk 1 above)
Comment (come-nc) [reply lib/private/EventSourceFactory.php:41]: It’s duplicating info from the OCP interface.
